sdks/python/apache_beam/runners/dataflow/internal/apiclient.py

```diff
@@ -447,6 +447,8 @@ def __init__(self, options, proto_pipeline):
     if self.google_cloud_options.labels:
       self.proto.labels = dataflow.Job.LabelsValue()
       labels = self.google_cloud_options.labels
+      if isinstance(labels, str):
+        labels = [labels]
```

Comment:
do you mean `options.view_as(GoogleCloudOptions).labels = {'env': 'bluecore-qa-extensibility', 'team': 'cdp-vertical'}`  is interpreted as a string although it is a Python dict?

Reply:
Whoops, meant to make this a draft until I full tested it. But you're right, this won't fully work yet. I think this should be more like:

```
if not isinstance(labels, Iterable):
   labels = [str(labels)]
```

But I'll fix/test this

Reply:
This should now handle both cases, and I added some representative test cases, PTAL